pass/disk.go

```diff
@@ -42,7 +47,55 @@ func defaultStorePath() (string, error) {
 	return filepath.EvalSymlinks(path)
 }
 
+// Set the configuration options for password matching.
+func (s *diskStore) SetConfig(path *string, use_fuzzy *bool) error {
+	if path != nil {
+		//todo validate path exists
+		s.path = *path
+	}
+	if use_fuzzy != nil {
+		s.useFuzzy = *use_fuzzy
+	}
+	return nil
+}
+
+// Do a search. Will call into the correct algoritm (glob, fuzzy (renstrom) or fuzzy (sahilm)
+// based on the settings present in the diskStore struct
 func (s *diskStore) Search(query string) ([]string, error) {
+	// default legacy glob search
+	if !s.useFuzzy {
+		fmt.Fprintf(os.Stderr, "Doing globsearch\n")
+		return s.GlobSearch(query)
+	} else {
+		fmt.Fprintf(os.Stderr, "Doing fuzzy\n")
+		return s.sahilmFuzzySerach(query)
+	}
+}
+
+// Fuzzy searches first get a list of all pass entries by doing a glob search
+// for the empty string, then apply appropriate logic to convert results to
+// a slice of strings, finally returning all of the unique entries.
+func (s *diskStore) sahilmFuzzySerach(query string) ([]string, error) {
+	var items []string
+	file_list, err := s.GlobSearch("")
+	if err != nil {
+		return nil, err
+	}
+
+	// The resulting match struct does not copy the strings, but rather
+	// provides the index to the original array. Copy those strings
+	// into the result slice
+	matches := sfuzzy.Find(query, file_list)
+	for _, match := range matches {
+		items = append(items, file_list[match.Index])
+	}
+
+	result := unique(items)
```

Comment:
GlobSearch already took care of duplicates, I don't think there could be duplicates anymore, so should be safe to remove